src/Symfony/Component/Config/Definition/ExpressionNode.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition;
+
+use Symfony\Component\Config\Definition\Exception\InvalidTypeException;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+/**
+ * This node represents a Symfony Expression Language Expression.
+ *
+ * @author Magnus Nordlander <magnus@fervo.se>
+ */
+class ExpressionNode extends VariableNode
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function validateType($value)
+    {
+        if (!$value instanceof Expression) {
+            $ex = new InvalidTypeException(sprintf(
+                'Invalid type for path "%s". Expected Symfony\\Component\\ExpressionLanguage\\Expression , but got %s.',
+                $this->getPath(),
+                gettype($value)
+            ));
+            if ($hint = $this->getInfo()) {
+                $ex->addHint($hint);
+            }
+            $ex->setPath($this->getPath());
+
+            throw $ex;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function preNormalize($value)
+    {
+        $value = parent::preNormalize($value);
+
+        return $this->transformToExpression($value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultValue()
+    {
+        $v = parent::getDefaultValue();
+
+        return $this->transformToExpression($v);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function isValueEmpty($value)
+    {
+        // a boolean value cannot be empty
+        return $value === null;
+    }
+
+    protected function transformToExpression($value)
+    {
+        if ($value == null) {
+            return;
+        }
+
+        if ($value instanceof Expression) {
+            return $value;
+        }
+
+        return new Expression($value);
```

Comment:
When the class is undefined because the component is missing, should we throw a meaningful exception before this?